src/cephadm/cephadm

```diff
@@ -4415,82 +4422,162 @@ def is_ipv6(address):
+            return True
+    return False
+
+
+def parse_mon_addrv(addrv_arg: str) -> List[EndPoint]:
+    """Parse mon-addrv param into a list of mon end points."""
     r = re.compile(r':(\d+)$')
-    base_ip = ''
+    addrv_args = []
+    addr_arg = addrv_arg
+    if addr_arg[0] != '[' or addr_arg[-1] != ']':
+        raise Error(f'--mon-addrv value {addr_arg} must use square backets')
+
+    for addr in addr_arg[1: -1].split(','):
+        hasport = r.findall(addr)
+        if not hasport:
+            raise Error(f'--mon-addrv value {addr_arg} must include port number')
+        port_str = hasport[0]
+        addr = re.sub(r'^v\d+:', '', addr)  # strip off v1: or v2: prefix
+        base_ip = addr[0:-(len(port_str)) - 1]
+        addrv_args.append(EndPoint(base_ip, int(port_str)))
+
+    return addrv_args
+
+
+def parse_mon_ip(mon_ip: str) -> List[EndPoint]:
+    """Parse mon-ip param into a list of mon end points."""
+    r = re.compile(r':(\d+)$')
+    addrv_args = []
+    hasport = r.findall(mon_ip)
+    if hasport:
+        port_str = hasport[0]
+        base_ip = mon_ip[0:-(len(port_str)) - 1]
+        addrv_args.append(EndPoint(base_ip, int(port_str)))
+    else:
+        # Use fixd ports for ceph monitor
+        addrv_args.append(EndPoint(mon_ip, 3300))
+        addrv_args.append(EndPoint(mon_ip, 6789))
+
+    return addrv_args
+
+
+def build_addrv_params(addrv: List[EndPoint]) -> str:
+    """Convert mon end points (ip:port) into the format: [v[1|2]:ip:port1]"""
+    addr_arg = ''
+    if len(addrv) == 1:
+        ip = addrv[0].ip
+        port = addrv[0].port
+        if port == 6789:
+            addr_arg = f'[v1:{ip}]'
+        elif port == 3300:
+            addr_arg = f'[v2:{ip}]'
+        else:
+            logger.warning(f'Using msgr2 protocol for unrecognized port {port}')
+            addr_arg = f'[v2:{ip}]'
+    elif len(addrv) == 2:
+        ip = addrv[0].ip
+        addr_arg = f'[v2:{ip}:3300,v1:{ip}:6789]'
+    else:
+        logger.warning(f'provided addrv {addrv} has a bad length')
+    return addr_arg
+
+
+def get_public_net_from_cfg(ctx: CephadmContext) -> Optional[str]:
+    """Get mon public network from configuration file."""
+    cp = read_config(ctx.config)
+    if not cp.has_option('global', 'public_network'):
+        return None
+
+    # Ensure all public CIDR networks are valid
+    public_network = cp.get('global', 'public_network')
+    rc, _, err_msg = check_subnet(public_network)
+    if rc:
+        raise Error(f'Invalid public_network {public_network} parameter: {err_msg}')
+
+    # Ensure all public CIDR networks are configured locally
+    configured_subnets = set([x.strip() for x in public_network.split(',')])
+    local_subnets = set([x[0] for x in list_networks(ctx).items()])
+    for net in configured_subnets:
+        if net not in local_subnets:
+            raise Error(f'The provided public CIDR network {net} is not configured locally.')
```

Comment:
can we also specify in the error message that this public network is from the provided config?

Reply:
I thought that word "configured" was enough. I'll add an explicit mention to something like `-c conf` to make it clear.